### PR TITLE
Add basic Airy function bindings

### PR DESCRIPTION
**Pull request by Vectornaut**
Function names are provisional. I've run informal tests against the SpecialFunctions package, but I need guidance on how to write formal tests.

These bindings only compute individual Airy functions and derivatives. It might be nice to add more bindings that take advantage of Arb's simultaneous computation feature.

I don't know how to preview the new docstrings, so they may have Markdown or LaTeX errors.

### CODE REVIEW

**Comment by thofma** [comment]
Is it possible to add some tests? Does not need to be complicated, just to make sure that the functions don't crash.

(@tthsqe12 @albinahlback What do you think of those names?)

**Review by tthsqe12 — CHANGES_REQUESTED** [review]
Names are great. Just need some test code to make sure the code doesn't crash and that it calls the right functions.

File: src/arb/arb.jl [hunk 1]
```diff
@@ -2002,6 +2002,64 @@ Return the number of partitions $p(n)$ as an element of $r$.
 """
 numpart(n::Int, r::ArbField) = numpart(fmpz(n), r)
 
+################################################################################
+#
+#  Hypergeometric and related functions
+#
+################################################################################
```
Comment (albinahlback) [inline src/arb/arb.jl:2009]: Maybe it is a good idea to have more sections within the file, but then I would expect the same for `acb.jl`.

**Comment by albinahlback** [comment]
Looks good to me, but of course needs some tests.

**Comment by albinahlback** [comment]
As for the tests: For `airy_ai` with `arb` arguments, I would go to about line 337 in `/test/arb/arb-test.jl` and append the respective tests, where the second argument in `overlaps` is the returned value when calculating the function at about twice the precision (100 bit precision should be just fine).

As Tommy implied (and correct me if I am wrong), the tests in Nemo are there only to ensure that the returned value is stable. It is Arb's task to ensure that they are correct.

**Comment by wbhart** [comment]
The tests in Nemo are currently pretty inadequate on the whole. I think they should test:

* that the binding are working (i.e. call the actual function at least once)
* corner cases
* that the documentation for Nemo matches what the function actually does
* DomainErrors and other exceptions are thrown as appropriate

Obviously if the function is not a straightforward wrapper and needs to do some computation, that should be tested as well as stuff in AbstractAlgebra.

But yes, it is Arb's responsibility to ensure the result is correct, assuming correct input is passed.

**Comment by thofma** [comment]
We will merge it as is and @tthsqe12 will add the missing tests.
